core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ApplyPreferredTableWriterPartitioning.java

```diff
@@ -57,6 +60,11 @@ public boolean isEnabled(Session session)
     @Override
     public Result apply(TableWriterNode node, Captures captures, Context context)
     {
+        if (getRetryPolicy(context.getSession()) == RetryPolicy.TASK && isFaultTolerantExecutionForcePreferredWritePartitioningEnabled(context.getSession())) {
+            // fault tolerant execution mitigates skew
```

Comment:
Maybe elaborate a bit on that. I don't think it's obvious.

Reply:
Added a comment